webapp/channels/src/components/new_search/search_box_type_selector.tsx

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {FormattedMessage} from 'react-intl';
+import {useSelector} from 'react-redux';
+import styled from 'styled-components';
+
+import {getLicense} from 'mattermost-redux/selectors/entities/general';
+
+import type {GlobalState} from 'types/store';
+
+const SearchTypeSelectorContainer = styled.div`
+    margin: 24px 32px 0px 24px;
+    display: flex;
+    align-items: center;
+    padding: 4px;
+    background-color: var(--center-channel-bg);
+    border-radius: var(--radius-m);
+    border: var(--border-default);
+    width: fit-content;
+    gap: 4px;
+`;
+
+type SearchTypeItemProps = {
+    selected: boolean;
+};
+
+const SearchTypeItem = styled.button<SearchTypeItemProps>`
+    display: flex;
+    cursor: pointer;
+    padding: 4px 10px;
+    background-color: ${(props) => (props.selected ? 'rgba(var(--button-bg-rgb), 0.08)' : 'transparent')};
+    color: ${(props) => (props.selected ? 'var(--button-bg)' : 'rgba(var(--center-channel-color-rgb), 0.75)')};
+    border-radius: 4px;
+    font-size: 12px;
+    line-height: 16px;
+    font-weight: 600;
+    border: none;
+    &:hover {
+        color: rgba(var(--center-channel-color-rgb), 0.88);
+        background: rgba(var(--center-channel-color-rgb), 0.08);
+    }
+`;
+
+type Props = {
+    searchType: string;
+    setSearchType: (searchType: string) => void;
+}
+
+const SearchTypeSelector = ({searchType, setSearchType}: Props) => {
+    const license = useSelector(getLicense);
+    let SearchPluginButtons = useSelector((state: GlobalState) => state.plugins.components.SearchButtons) || [];
+    if (license.IsLicensed !== 'true') {
+        SearchPluginButtons = [];
+    }
+    return (
+        <SearchTypeSelectorContainer>
+            <SearchTypeItem
+                selected={searchType === 'messages'}
+                onClick={() => setSearchType('messages')}
+            >
+                <FormattedMessage
+                    id='search_bar.usage.search_type_messages'
+                    defaultMessage='Messages'
+                />
+            </SearchTypeItem>
+            <SearchTypeItem
+                selected={searchType === 'files'}
+                onClick={() => setSearchType('files')}
+            >
+                <FormattedMessage
+                    id='search_bar.usage.search_type_files'
+                    defaultMessage='Files'
+                />
+            </SearchTypeItem>
+            {SearchPluginButtons.map(({component, pluginId}: any) => {
+                const Component = component as React.ComponentType;
+                return (
+                    <SearchTypeItem
+                        key={pluginId}
+                        selected={searchType === pluginId}
+                        onClick={() => setSearchType(pluginId)}
```

Comment:
In order to avoid a new function on every render, you may want to move this to a different component. Something like...
```
const PluginSearchTypeItem = ({pluginId, searchType, setSearchType, chilren}) => {
    const onClick = useCallback(() => setSearchType(pluginId));
    return (
         <SearchTypeItem
             key={pluginId}
             selected={searchType === pluginId}
             onClick={onClick}
         >
             {children}
         </SearchTypeItem>
    );
```

Reply:
This one, like the other one, gets me into more complicated code and I don't want to do it until we see we need to gain that performance.